### ISSUE TITLE
Duplicate Code warning occurs for conditional imports even with ignore-imports=yes 

### ISSUE THREAD

**Issue by dpinol** [pylint-dev/pylint#8914]
### Bug description

If imports are within an if branch (eg. if typing.TYPE_CHECKING), `--ignore-imports=y` does not ignore duplicated imports.

### Steps to reproduce

 Define the following three files:

a.py

```python
""" Mandatory Docstring """


class A:
    pass


class B:
    pass


class C:
    pass


class D:
    pass
```

b.py

```python
""" Mandatory Docstring """
from __future__ import annotations

import typing

if typing.TYPE_CHECKING:
    from a import (
        A,
        B,
        C,
        D,
    )


def f(_d: D, _b: B, _a: A, _c: C) -> None:
    pass
```

c.py

```python
""" Mandatory Docstring """
from __future__ import annotations

import typing

if typing.TYPE_CHECKING:
    from a import (
        A,
        B,
        C,
        D,
    )


def g(_a: A, _b: B, _c: C) -> D:
    pass
```

    1. Place them in an empty folder.
    2. Run pylint over them, using the ignore-imports option: `pylint --ignore-imports=yes *.py`

### Command used

```shell
pylint --ignore-imports=y b.py c.py
```

### Pylint output

```shell
************* Module c
c.py:1:0: R0801: Similar lines in 2 files
==b:[5:16]
==c:[5:16]
if typing.TYPE_CHECKING:
    from a import (
        A,
        B,
        C,
        D,
    )


def f(_d: D, _b: B, _a: A, _c: C) -> None:
    pass (duplicate-code)

------------------------------------------------------------------
Your code has been rated at 9.17/10 (previous run: 9.17/10, +0.00)
```

### Expected behavior

The duplicate lines are import statements that have been wrapped onto multiple lines. They should not trigger the similarity checker when the `--ignore-imports=yes` option is used.

Also, I don't get why pylint complains about also about lines 15-16, since functions in b.y & c.py are quite different.

### Pylint version

```shell
No config file found, using default configuration
pylint 2.17.5
astroid 2.15.6
Python 3.11.3 (main, May  2 2023, 16:42:49) [GCC 11.3.0]
```

### OS / Environment

ubuntu
